dotnet/src/SemanticKernel.Abstractions/Search/TextSearchResult.cs

```diff
@@ -0,0 +1,40 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System.Diagnostics.CodeAnalysis;
+using System.Text.Json.Serialization;
+
+namespace Microsoft.SemanticKernel.Search;
+
+/// <summary>
+/// Represents a search result retrieved from a <see cref="KernelSearchResults{T}" /> instance.
+/// </summary>
+/// <remarks>
+/// Initializes a new instance of the <see cref="TextSearchResult"/> class.
+/// </remarks>
+[Experimental("SKEXP0001")]
+public class TextSearchResult(string? name, string? value, string? link, object? innerContent)
```

Comment:
It looks like `link` and `innerContent` should be optional. Also, since there is no coupling to `KernelContent`, maybe it makes sense to rename `innerContent` to `innerResult`.